app/controllers/admin/users_controller.rb

```diff
@@ -16,6 +16,12 @@ def create
 
   def update
     user = User.find(params[:id])
+    if params[:upgrade]
+      User.find(params[:id]).update(:role => 1)
```

Comment:
Minor, but since you already found the user via params[:id] and saved it to the `user` variable, you could just change this to `user.update` instead of re-finding. Also I think you could probably use symbol notation instead of hash rocket? That's not a big deal though. Aka `role: 1`.

Reply:
Ah see thats what I thought too, but it just changes the role for that variable name, and doesn't actually change the role of the user in the database. Thats why in the test I change the user to updated_user because user_1 was just a snapshot of the user before the test. Am I right? Or just way off?

Reply:
Oh interesting! That makes sense!